File: RB/cpp/RiotBase/Variable/Symbol/Executed.cpp
#include "../../System/Scast.h"
#include "Executed.h"

namespace RiotBase {

std::string Executed::type_name() {
    return "Executed";
}

Rp **Executed::prox_ptr() {
    return &val.rp;
}

const Rp *Executed::prox_ref() const {
    static_assert( sizeof( Executed ) == sizeof( Lazyfier ), "..." );
    return val.rp;
}

void Executed::write_structure( std::ostream &os ) const {
    RO_CALL( write_structure, val.rp, os << "Executed(" ); os << ')';
}

bool Executed::is_symbolic() const {
    return RO_CALL( is_symbolic, val.rp );
}

} // namespace RiotBase

File: RB/ts/RiotBase/Variable/Symbol/Executed.ts
import Variable from "../Core/Variable";
import Rp       from "../Core/Rp";


//
export default
class Executed extends Rp {
    prox_ref() : Variable {
        return this.sym;
    }

    prox_ptr() : Variable {
        return this.sym;
    }

    sym        : Variable;
}
Rp.make_templ( Executed );
